sfs/time/drivingfunction.py

```diff
@@ -255,3 +256,194 @@ def apply_delays(signal, delays):
     for column, row in enumerate(delays_samples):
         out[row:row + len(data), column] = data
     return util.DelayedSignal(out, samplerate, offset_samples / samplerate)
+
+
+def nfchoa_25d_plane(x0, r0, npw, max_order=None, c=None, fs=44100, normalize=True):
+    """Vritual plane wave by 2.5-dimensional NFC-HOA.
+
+    Parameters
+    ----------
+    x0 : (N, 3) array_like
+        Sequence of secondary source positions.
+    r0 : float
+        Radius of the circular secondary source distribution
+    npw : (3,) array_like
+        Unit vector (propagation direction) of plane wave.
+    max_order : int
+        Ambisonics order
+    c : float, optional
+        Speed of sound
+
+    Returns
+    -------
+    delay : float
+        Overall delay (common to all secondary source)
+    weight : float
+        Overall weight (common to all secondary sources)
+    sos : list
+        Second-order section filters
+    phaseshift : float
+        Phase shift
+
+    References
+    ----------
+        S. Spors, V. Kuscher, J. Ahrens (2011) - "Efficient realization of
+       model-based rendering for 2.5-dimensional near-field compensated higher
+       order Ambisonics", WASPAA, p. 61-64
+
+       See Eq.(10)
+    """
+    max_order = util.max_order_circular_harmonics(len(x0), max_order)
+    if c is None:
+        c = defs.c
+
+    x0 = util.asarray_of_rows(x0)
+    npw = util.asarray_1d(npw)
+    phipw, _, _ = util.cart2sph(*npw)
+    phi0, _, _ = util.cart2sph(*x0.T)
+
+    delay = -r0/c
+    weight = 2
+    sos = []
+    for m in range(0, max_order+1):
+        _, p, _ = besselap(m, norm='delay')
+        # TODO: modify "besselap" for very high orders (>150)
+        s0 = np.zeros(m)
+        sinf = (c/r0)*p
+        z0 = np.exp(s0/fs)
```

Comment:
It's unclear to me why you create a bunch of zeros and then divide them by something.
And the exponential of that will always be 1, won't it?
This can't be right?!?

Reply:
You're right. `z0` will be just ones. Will be fixed.

I just wanted to make it easier to understand what's going on (at least for myself). The poles/zeros in the Laplace domain are mapped to the z-domain using the matches z-transform. I thought this might also help reviewing the code.

Reply:
OK, if you want to show this, you should add a comment. Otherwise it looks like a mistake.

Or you could make a separate (private?) function with a meaningful name that should also make clear what's happening?

You could probably re-use such a function in `nfchoa_25d_point()`?